src/step-9/routes/login.js

```diff
@@ -0,0 +1,24 @@
+import errors from 'http-errors'
+import S from 'fluent-json-schema'
+
+const schema = {
+  body: S.object()
+    .prop('username', S.string().required())
+    .prop('password', S.string().required()),
+  response: {
+    200: S.object().prop('token', S.string().required()),
+  },
+}
+
+export default async function login(fastify) {
+  fastify.post('/login', { schema }, async req => {
+    const { username, password } = req.body
+
+    // sample auth check
+    if (username !== password) {
+      throw errors.Unauthorized()
+    }
```

Comment:
should this be here? step 9 should be the authenticate plugin

Reply:
![Screenshot 2021-02-02 at 17 10 25](https://user-images.githubusercontent.com/272483/106627940-9169a480-6579-11eb-8640-37a2977bf3e3.png)